client/signup/jetpack-connect/index.jsx

```diff
@@ -174,16 +171,18 @@ const JetpackConnectMain = React.createClass( {
 		}
 
 		if ( this.state.waitingForSites && ! this.props.isRequestingSites ) {
+			// FIXME: Can this be improved?
```

Comment:
My point (that I should probably have spent some more time explaining) was that this can be avoided, but it would require some refactoring. Let's say that with the way it's currently implemented, the inner state variable for delaying the call is a good way to achieve this.

@sirreal we're doing this because the sites request can take long sometimes, especially for users with a lot of sites. And we don't want to block the connect page until the sites are loaded. 

In a better implementation it would all be stored in Redux, and we'd queue up the site checking there; then when the sites are loaded, the checking in the queue would be executed automatically.

Let's conclude this one - refactoring it will definitely go outside of this PR's scope. I only commented there because of your FIXME comment.